vyper/semantics/validation/annotation.py

```diff
@@ -47,8 +47,8 @@ class StatementAnnotationVisitor(_AnnotationVisitorBase):
         vy_ast.Raise,
     )
 
-    def __init__(self, fn_node: vy_ast.FunctionDef, namespace: dict) -> None:
-        self.func = fn_node._metadata["type"]
+    def __init__(self, namespace: dict, fn_node: vy_ast.FunctionDef = None) -> None:
```

Comment:
why can `fn_node` be `None` now? I think other code might depend on it being there.

Reply:
This change was made to facilitate the annotation of the AST for `sqrt` builtin, which does not contain a `FunctionDef` node. I will add a check for `self.func` in the instances where it is used downstream.